docker/wrapper.py

```diff
@@ -12,37 +12,131 @@
-        f.write(generate_config(args.star, args.rsem, args.kallisto, mount,
+        f.write(generate_config(args.star, args.rsem, args.kallisto, args.hera, output_dir,
                                 args.disable_cutadapt, args.save_bam, args.save_wiggle,
-                                args.bamqc))
+                                args.bamqc, args.max_sample_size))
     loglevel = log.getEffectiveLevel()
 
-    command = ['toil-rnaseq', 'run',
+    if args.auto_scale:
+        #Pick the working directory carefully; it must exist on the workers
+        #Do not let Toil pick the directory because this will not work when
+        #launching the workflow with Dockstore because dockstore uses cwltool
+        #which makes the file system read only which will prevent Toil from 
+        #creating the directory. We cannot use a directory like
+        #/home/ubunut/... becuase that does not exist on coreos which is what
+        #the worker nodes use for an OS; '/tmp' will usually already be 
+        #created on most Linux distributions
+        command = ['toil-rnaseq', 'run',
+               job_store,
+               '--config', config_path,
+               '--workDir', '/tmp',
+               args.toilLoggingOption,
+               '--retryCount', '1000000']
+    else:
+        command = ['toil-rnaseq', 'run',
                job_store,
                '--config', config_path,
                '--workDir', work_dir,
                args.toilLoggingOption,
                '--retryCount', '1']
+
+
     if args.resume:
         command.append('--restart')
     if args.cores:
         command.append('--maxCores={}'.format(args.cores))
-    path_to_manifest = generate_manifest(args.sample_tar, args.sample_single, args.sample_paired, work_dir, args.output_basename)
+    path_to_manifest = generate_manifest(args.sample_tar, args.sample_single, \
+        args.sample_paired, work_dir, args.output_basenames)
     command.append('--manifest=' + path_to_manifest)
+
+    if args.auto_scale:
+        #get the private ip address of the VM on which the container is running
+        #this probably requires the --net=host option with the docker run
+        #commmand. This will be where the mesos master is running.
+        private_ip_address = socket.gethostbyname(socket.gethostname())
+        #add the extra options to run autoscaling to the command to run the pipeline
+        autoscale_options = ('--provisioner ' +  args.provisioner + ' --nodeType ' \
+            + args.node_type + ' --batchSystem mesos --maxNodes ' + str(args.max_nodes) \
+            + ' --mesosMaster ' + private_ip_address + ':5050  --logLevel DEBUG')
+        autoscale_options_list = autoscale_options.split()
+        command.extend(autoscale_options_list)
+
+        if args.provisioner == 'aws':
+            os.environ['AWS_ACCESS_KEY_ID'] = args.credentials_id
+            os.environ['AWS_SECRET_ACCESS_KEY'] = args.credentials_secret_key
+        '''
```

Comment:
Remove comment block (unless it needs to be here?)

Reply:
I couldn't decide if we needed the option to provide a credentials file instead of credentials in separate arguments. We have to have separate arguments for Consonance support. I guess separate arguments is enough.